File: app/src/main/java/com/example/beerhive/beerlist/BeerContract.java
package com.example.beerhive.beerlist;

import com.example.beerhive.network.model.BeerResponse;

import java.util.List;

public interface BeerContract {
    interface BeerView {
        void displayBeerList(List<BeerResponse> responseList);

        void showProgressDialog();

        void showErrorMesage();

        void dismissProgressDialog();
    }

    interface UserActionsListener {
        void loadBeerList();
    }
}

File: app/src/main/java/com/example/beerhive/beerlist/BeerContract.kt
package com.example.beerhive.beerlist

import com.example.beerhive.network.model.BeerResponse

interface BeerContract {
    interface BeerView {
        fun displayBeerList(responseList: List<BeerResponse?>?)
        fun showProgressDialog()
        fun showErrorMessage()
        fun dismissProgressDialog()
    }

    interface UserActionsListener {
        fun loadBeerList()
    }
}